File: server/router/z_teacher.go
package router

import (
	"gin-vue-admin/api/v1"
	"gin-vue-admin/middleware"
	"github.com/gin-gonic/gin"
)

// InitTeacherRouter 初始化 Teacher 路由信息
func InitTeacherRouter(Router *gin.RouterGroup) {
	TeacherRouter := Router.Group("teacher").Use(middleware.OperationRecord())
	{
		TeacherRouter.POST("createTeacher", v1.CreateTeacher)   // 新建Teacher
		TeacherRouter.DELETE("deleteTeacher", v1.DeleteTeacher) // 删除Teacher
		TeacherRouter.DELETE("deleteTeacherByIds", v1.DeleteTeacherByIds) // 批量删除Teacher
		TeacherRouter.PUT("updateTeacher", v1.UpdateTeacher)    // 更新Teacher
		TeacherRouter.GET("findTeacher", v1.FindTeacher)        // 根据ID获取Teacher
		TeacherRouter.GET("getTeacherList", v1.GetTeacherList)  // 获取Teacher列表
		TeacherRouter.GET("getAllTeachers", v1.GetAllTeachers)  // 获取所有Teacher列表
	}
}

File: web/src/api/z_teacher.js
import service from '@/utils/request'

// @Tags Teacher
// @Summary 创建Teacher
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Teacher true "创建Teacher"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /teacher/createTeacher [post]
export const createTeacher = (data) => {
    return service({
        url: '/teacher/createTeacher',
        method: 'post',
        data
    })
}

// @Tags Teacher
// @Summary 删除Teacher
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Teacher true "删除Teacher"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /teacher/deleteTeacher [delete]
export const deleteTeacher = (data) => {
    return service({
        url: '/teacher/deleteTeacher',
        method: 'delete',
        data
    })
}

// @Tags Teacher
// @Summary 删除Teacher
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.IdsReq true "批量删除Teacher"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"删除成功"}"
// @Router /teacher/deleteTeacher [delete]
export const deleteTeacherByIds = (data) => {
    return service({
        url: '/teacher/deleteTeacherByIds',
        method: 'delete',
        data
    })
}

// @Tags Teacher
// @Summary 更新Teacher
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Teacher true "更新Teacher"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"更新成功"}"
// @Router /teacher/updateTeacher [put]
export const updateTeacher = (data) => {
    return service({
        url: '/teacher/updateTeacher',
        method: 'put',
        data
    })
}

// @Tags Teacher
// @Summary 用id查询Teacher
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body model.Teacher true "用id查询Teacher"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"查询成功"}"
// @Router /teacher/findTeacher [get]
export const findTeacher = (params) => {
    return service({
        url: '/teacher/findTeacher',
        method: 'get',
        params
    })
}

// @Tags Teacher
// @Summary 分页获取Teacher列表
// @Security ApiKeyAuth
// @accept application/json
// @Produce application/json
// @Param data body request.PageInfo true "分页获取Teacher列表"
// @Success 200 {string} string "{"success":true,"data":{},"msg":"获取成功"}"
// @Router /teacher/getTeacherList [get]
export const getTeacherList = (params) => {
    return service({
        url: '/teacher/getTeacherList',
        method: 'get',
        params
    })
}